sources/packages/backend/libs/integrations/src/services/student-loan-balance/student-loan-balance.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from "@nestjs/common";
+import { SystemUsersService } from "@sims/services";
+import { StudentLoanBalance } from "@sims/sims-db";
+import { EntityManager } from "typeorm";
+
+@Injectable()
+export class StudentLoanBalanceService {
+  constructor(private readonly systemUserService: SystemUsersService) {}
+
+  /**
+   * Most recent balance present on database.
+   * If no records are present, returns undefined.
+   * @param referenceBalanceDate used as a reference to get the last balance before this date.
+   * @param entityManager used to get the data in and share the database transaction.
+   * @returns the most updated balance date, if present, otherwise, undefined.
+   */
+  async getLastBalanceDate(
+    referenceBalanceDate: Date,
+    entityManager: EntityManager,
+  ): Promise<Date | undefined> {
+    const maxBalanceDateAlias = "maxBalanceDate";
+    const lastBalanceDate = await entityManager
```

Comment:
I would recommend to have it typed instead of type any.

![image](https://github.com/bcgov/SIMS/assets/54600590/3e589f38-a22d-40f3-ac41-57827ada924f)

Reply:
It does not work!

Reply:
Got it after our chat. The code is updated.